core/src/main/scala/cats/instances/seq.scala

```diff
@@ -127,7 +127,23 @@ trait SeqInstances extends cats.kernel.instances.SeqInstances {
       }
 
       final override def traverse[G[_], A, B](fa: Seq[A])(f: A => G[B])(implicit G: Applicative[G]): G[Seq[B]] =
-        G.map(Chain.traverseViaChain(fa.toIndexedSeq)(f))(_.toVector)
+        G match {
+          case x: StackSafeMonad[G] =>
+            x.map(Traverse.traverseDirectly(fa)(f)(x))(_.toList)
+          case _ =>
+            G.map(Chain.traverseViaChain(fa.toIndexedSeq)(f))(_.toVector)
+        }
+
+      override def traverse_[G[_], A, B](fa: Seq[A])(f: A => G[B])(implicit G: Applicative[G]): G[Unit] =
+        G match {
+          case x: StackSafeMonad[G] => Traverse.traverse_Directly(fa)(f)(x)
+          case _ =>
+            foldRight(fa, Always(G.pure(()))) { (a, acc) =>
```

Comment:
```suggestion
            foldRight(fa, Eval.now(G.unit)) { (a, acc) =>
```

Reply:
😆 the `Map` instance. Sorry, the conversation has gotten so long that it's hard to follow. I think we agreed upon only benchmarking `Chain`, `Vector` and `List` so I said that I'd update the `Map` instance once we've decided what implementation to use for all the other collections.